p360monitoring.php

```diff
@@ -0,0 +1,245 @@
+            'FriendlyName' => Translator::translate('p360monitoring_label_additional_websites'),
+            'Type' => 'text',
+            'Size' => '25',
+            'Default' => '',
+            'SimpleMode' => true,
+        ],
+    ];
+}
+
+function p360monitoring_ClientArea(array $params)
+{
+    $keyId = $params['customfields']['keyId'];
+    $uid = $params['customfields']['activationInfoUid'];
+    $activated = $params['customfields']['activationInfoActivated'];
+    $activationLink = $params['customfields']['activationLink'];
+    $terminated = $params['model']->serviceProperties->get('terminated');
+    $suspended = $params['model']->serviceProperties->get('suspended');
+    $servers = Params::serverAllowance($params);
+    $websites = Params::websiteAllowance($params);
+
+    try {
+        if (!$activated) {
+            $kaApi = new KaApi($params);
+            $license = $kaApi->retrieveLicense($keyId);
+            $activationLink = $license['keyIdentifiers']['activationLink'];
+            $activated = $license['activationInfo']['activated'];
+            $uid = $license['activationInfo']['uid'];
+            $terminated = $license['terminated'];
+            $suspended = $license['suspended'];
+
+            $params['model']->serviceProperties->save([
+                'activationLink' => $activationLink,
+                'activationInfoUid' => $uid,
+                'activationInfoActivated' => $activated,
+                'terminated' => $terminated,
+                'suspended' => $suspended
+            ]);
+        }
+
+        $returnHtml = '<div class="tab-content"><div class="row"><div class="col-sm-1"><strong>' . Translator::translate('p360monitoring_label_features') . '</strong></div></div><div class="row"><div class="col-sm-1">' . Translator::translate('p360monitoring_label_servers') . '</div><div class="col-sm-3 text-right">' . $servers . '</div></div><div class="row"><div class="col-sm-1">' . Translator::translate('p360monitoring_label_websites') . '</div><div class="col-sm-3 text-right">' . $websites . '</div></div>';
+
+        if ($activated) {
+            $returnHtml .= '<div class="row"><div class="col-sm-3 text-left">' . Translator::translate('p360monitoring_button_license_activated') . '</div></div>';
+        }
+
+        $returnHtml .= '</div><br/>';
+
+        if (!$terminated && !$suspended) {
+            if (!$activated) {
+                $returnHtml .= '<div class="tab-content"><a class="btn btn-block btn-info" href="' . $activationLink . '" target="_blank">' . Translator::translate('p360monitoring_button_activate_license') . '</a></div><br/>';
+            }
+
+            $returnHtml .= '<div class="tab-content"><a class="btn btn-block btn-default" href="https://monitoring.platform360.io/dashboard/overview" target="_blank">' . Translator::translate('p360monitoring_button_dashboard') . '</a></div><br/>';
+        }
+
+        return $returnHtml;
+    } catch (Exception $e) {
```

Comment:
Depends on the PHP version you should use `Throwable`

Reply:
Fixed